### PR TITLE
.NET 8

### PR DESCRIPTION
**Pull request by TheTedder**
I took the liberty of upgrading to .NET 8 since .NET 7 is out of support. I also updated almost all of our dependencies. This is bound to break something so it's a good idea to test this heavily. The default placeholder JWT private key had to be increased in length because it didn't have enough entropy or something.

### CODE REVIEW

**Comment by zysim** [comment]
1. Gotta allow anonymous fetching of Categories. We really don't need to be authorised for that
2. How do you generate `openapi.json` again? It's been so long

File: LeaderboardBackend/Models/Entities/ApplicationContext.cs [hunk 1]
```diff
@@ -31,9 +32,22 @@ public ApplicationContext(DbContextOptions<ApplicationContext> options)
     public void MigrateDatabase()
     {
         Database.Migrate();
+        bool tempConnection = false;
+        NpgsqlConnection connection = (NpgsqlConnection)Database.GetDbConnection();
```
Comment (zysim) [inline LeaderboardBackend/Models/Entities/ApplicationContext.cs:36]: When does the connection drop?

Comment (TheTedder) [reply LeaderboardBackend/Models/Entities/ApplicationContext.cs:36]: I don't understand the question. Can you rephrase?

**Comment by zysim** [comment]
Will do more testing tomorrow

**Comment by TheTedder** [comment]
> 1. Gotta allow anonymous fetching of Categories. We really don't need to be authorised for that
> 
>     2. How do you generate `openapi.json` again? It's been so long
> 
> 
> Also I've done some testing of the API. Not like "running tests" testing, but calling the API itself, locally. Able to register a new user, log in/out, and create leaderboards and categories, but only if we're authorised. So far so good.
> 
> All tests pass, too.

This PR doesn't touch category logic. Also, we're gonna go through and change all that anyway after account-related flows are done.

**Comment by zysim** [comment]
> > 1. Gotta allow anonymous fetching of Categories. We really don't need to be authorised for that
> 
> This PR doesn't touch category logic. Also, we're gonna go through and change all that anyway after account-related flows are done.

Yeah that's fair. I'll add a ticket for that if there isn't one, already.

**Comment by zysim** [comment]
Yeah the backend works. Only problems are some action behaviours. I'll list them out in a separate ticket.

**Review by zysim — APPROVED** [review]
